docs/cep/index.md

```diff
@@ -22,11 +22,111 @@ You can process streams to perform the following actions with your data:
 - Derive insights by identifying event patterns in data streams.
 - Summarize data with time windows and incremental aggregations.
 - Extract, transform, and load (ETL) collections, tailing files, and scraping HTTP endpoints.
 - Integrating stream data and trigger actions based on the data. This can be a single service request or a complex enterprise integration flow.
+- [Consume](query-guide/stream.md#source) and [publish](query-guide/stream.md#sink) events.
+- Run premade and custom [functions](query-guide/functions/index.md).
+- Write custom [JavaScript functions](query-guide/script.md) to interact with your streams.
+- Query, modify, and join the data stored in [tables](query-guide/table-collection.md) which support primary key constraints and indexing.
+- Rule processing based on single event using [`filter`](query-guide/query.md#filter) operator, `if-then-else` and `match` [functions](query-guide/query.md#function), and many others.
+
+These features allows you to build robust global data processing and integration pipelines at the edge by combining powerful stream processing, multi-model database and geo-replicated streams capabilities.
+
+:::tip
+"Have different business use cases in separate stream workers."
+This is recommended as it allows users to selectively deploy the applications based on their business needs. It is also recommended to move the repeated stream processing logic that exists in multiple stream workers, such as message retrieval and preprocessing, to a common stream workers, whereby reducing code duplication and improving maintainability. In this case, to pass the events from one Stream App to another, configure them using a common stream or collection using `stream` Sink and `stream` Source.
+
+Stream workers provide an isolated execution environment for your processing logic that allows you to deploy and execute processing logic independent of other stream workers in the system. Therefore, it's always recommended to have processing logic related to a single use case in a single StreamApp. This will help you to group processing logic and easily manage addition and removal of various use cases.
+:::
+
+
+## Stream Worker Explanation
+
+This section explains the parts of this stream worker and what they are doing.
+
+### Metadata
+
+This information defines basic information about the stream worker. Every stream worker must have at least a name and query language version in order to be valid.
+
+- **Name** - `@App:name("ExampleApp")`
+- **Query language version (optional)** - @App:qlVersion("2")
+- **Description (optional)** - @App:description('An application for enriching transactions.')
+- **Other information (optional)** - By convention, you can enter a comment with testing information, update logs, or other useful information at the beginning of the stream worker definition between `/**` and `**/`. This is similar to a docstring in functions.
+
+### Input and Output
```

Comment:
Note from @diana-macrometa:

Let's narrow this a bit for now, just link to the Sink and Source sections, because they are the only two that are actually required. Table won't always be present, as I understand it. (ARGH, we need to get with @StoyanTc on this.)